src/category_theory/closed/cartesian.lean

```diff
@@ -339,21 +349,13 @@ def cartesian_closed_of_equiv (e : C ≌ D) [h : cartesian_closed C] : cartesian
       apply nat_iso.of_components _ _,
       intro Y,
       { apply as_iso (prod_comparison e.inverse X (e.functor.obj Y)) ≪≫ _,
-        exact ⟨limits.prod.map (𝟙 _) (e.unit_inv.app _),
-              limits.prod.map (𝟙 _) (e.unit.app _),
-              by simpa [←prod_map_id_comp, prod_map_id_id],
-              by simpa [←prod_map_id_comp, prod_map_id_id]⟩, },
+        apply prod.map_iso (iso.refl _) (e.unit_iso.app Y).symm },
       { intros Y Z g,
-        simp only [prod_comparison, inv_prod_comparison_map_fst, inv_prod_comparison_map_snd,
-          prod.lift_map, functor.comp_map, prod_functor_obj_map, assoc, comp_id,
-          iso.trans_hom, as_iso_hom],
-        apply prod.hom_ext,
-        { rw [assoc, prod.lift_fst, prod.lift_fst, ←functor.map_comp,
-            limits.prod.map_fst, comp_id], },
-        { rw [assoc, prod.lift_snd, prod.lift_snd, ←functor.map_comp_assoc, limits.prod.map_snd],
-          simp only [iso.hom_inv_id_app, assoc, equivalence.inv_fun_map,
-            functor.map_comp, comp_id],
-          erw comp_id, }, },
+        dsimp [prod_comparison],
+        simp [prod.comp_lift, ← e.inverse.map_comp, ← e.inverse.map_comp_assoc],
+          -- I wonder if it would be a good idea to make `map_comp` a simp lemma the other way round
```

Comment:
I suspect that there's an initial period when setting up the basics (i.e. lemmas about functors) that `map_comp` is a good simp lemma, and then at some point the tide turns, and `←map_comp` becomes more useful!

I'm not sure how to test / achieve / document this.

Reply:
I had a go at changing `map_comp` to be the other way round, and tried propagating that change through mathlib a couple of days ago; after not very long I came to basically this conclusion as well, so I certainly believe this is true. I'm also in agreement that it's not clear how to achieve this.